### DISCUSSION TITLE
General questions

### DISCUSSION THREAD

**Discussion by KWDLibra** [marda-alliance/metadata_extractors#20] [Q&A] [answered]
Howdy all, I just had some preliminary questions from my short exploration of the API extractor tool:

1. Is the output limited to one dimensional data sets? Any way to get multi dimensional?
2. Right now it is using .mpr files, is there a planned list of other supported file types or data formats?

**Comment by ml-evs** [comment, marked as answer]
Hi @KWDLibra, to take your questions in turn:

1. There's no limitation to what an extractor can return (we have not made any progress defining extractor output schemas yet, and probably won't in our initial version unless we get significant extra buy-in/interest). For now, extractors can return either raw stdlib Python objects (e.g., dicts, JSON strings etc) or pandas dataframes and xarray datasets (which requires pandas/xarray to be installed in the host environment).
2. Part of the focus on this project has been simply tabulating file formats and the extractors that can be used on them, in a programmatic way. We have done this in the registry repo (https://github.com/marda-alliance/metadata_extractors_registry) with programmatic API currently hosted at https://marda-registry.fly.dev/filetypes. If there are file types you work with and are interested in, feel free to raise an issue there! As it stands, you can use the marda extractors Python API to automatically call [`yadg`](https://github.com/dgbowl/yadg) (currently the only contributed extractor) on any file type it supports, e.g., mpr, xrdml and various others.

We're still very much at the proof-of-concept phase I would say, and are definitely soliciting contributions of extractor codes and file types before an initial release (that we will try to write a paper on through MaRDA). If you're interested in getting involved feel free to shoot me an email (address is on my website linked from my profile) and I can add you to the mailing list.

**Reply by KWDLibra** [reply]
Thank you very much!
